kyo-core/shared/src/main/scala/kyo/Timer.scala

```diff
@@ -1,288 +1,213 @@
 package kyo
 
-import java.util.concurrent.*
-import kyo.scheduler.util.Threads
-
-/** A timer for scheduling tasks to run after a delay or periodically. */
-abstract class Timer:
-    def unsafe: Timer.Unsafe
+/** A utility for scheduling recurring tasks with different timing strategies.
+  *
+  * Timer provides two main scheduling approaches:
+  *   - `repeatWithDelay`: Executes tasks with a fixed delay between task completions
+  *   - `repeatAtInterval`: Executes tasks at fixed time intervals, regardless of task duration
+  */
+object Timer:
```

Comment:
Do we need a distinct object for `Timer`? I generally prefer extension methods (or at least avoiding too many distinct objects) for discoverability.

I think most other effect systems don't have a concept of 'Timer'

Reply:
Yeah, it doesn't feel necessary. Perhaps the methods could be in `Clock` since `sleep` is already there?

Reply:
I've made the change, it seems ok